src/SExpr.purs

```diff
@@ -77,6 +77,15 @@ data ListRestPattern
    | PListEnd
    | PListNext Pattern ListRestPattern
 
+type Comment = List CommentElem
```

Comment:
Thinking more about naming, maybe the `Maybe` could be pushed into the synonym? So rather than `DocComment`, maybe `DocOpt`,  defined as `Maybe (List DocElem)`? (Or similar..)